File: frontend/src/api/image.ts
import http from './http'

export interface ImageListItem {
  id: string
  title: string
  image_file_path?: string
  style?: string
  status: string
  created_at: string
}

export interface ImageDetail {
  id: string
  title: string
  original_input: string
  prompt_cn?: string
  prompt_en?: string
  image_file_path?: string
  style?: string
  aspect_ratio?: string
  status: string
  created_at: string
  updated_at: string
}

export function getImages(page = 1, pageSize = 20): Promise<ImageListItem[]> {
  return http.get('/images', { params: { page, page_size: pageSize } })
}

export function getImage(id: string): Promise<ImageDetail> {
  return http.get(`/images/${id}`)
}

export function deleteImage(id: string): Promise<void> {
  return http.delete(`/images/${id}`)
}

File: backend/app/models/image.py
"""图片模型"""

import uuid
from datetime import datetime
from sqlalchemy import Column, String, Text, DateTime
from app.database import Base


class Image(Base):
    __tablename__ = "images"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    title = Column(String(200), nullable=False, comment="标题")
    original_input = Column(Text, nullable=False, comment="用户原始描述")
    prompt_cn = Column(Text, nullable=True, comment="中文描述")
    prompt_en = Column(Text, nullable=True, comment="英文生成 Prompt")
    image_file_path = Column(String(500), nullable=True, comment="图片文件路径")
    style = Column(String(50), nullable=True, comment="图片风格")
    aspect_ratio = Column(String(10), nullable=True, comment="宽高比")
    status = Column(String(20), nullable=False, default="completed", comment="状态")
    created_at = Column(DateTime, default=datetime.now)
    updated_at = Column(DateTime, default=datetime.now, onupdate=datetime.now)
